### PR TITLE
Add missing classes for the jgit native build.

### PR DESCRIPTION
**Pull request by andrejpetras**
I add the missing classes for the jgit native build.

Test application: https://github.com/lorislab/fluh

### CODE REVIEW

**Comment by gsmet** [comment]
@gastaldi can you have a look?

It might be something I could include in 0.23.2 tomorrow if we decide to merge early tomorrow.

File: extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java [hunk 1]
```diff
@@ -29,7 +29,59 @@ ReflectiveClassBuildItem reflection() {
...
+                "com.jcraft.jsch.jce.ARCFOUR128",
+                "com.jcraft.jsch.jce.ARCFOUR256",
+                "com.jcraft.jsch.UserAuthNone",
+                "com.jcraft.jsch.UserAuthPassword",
+                "com.jcraft.jsch.UserAuthKeyboardInteractive",
+                "com.jcraft.jsch.UserAuthPublicKey",
+                "com.jcraft.jsch.UserAuthGSSAPIWithMIC",
+                "com.jcraft.jsch.jgss.GSSContextKrb5",
+                "com.jcraft.jsch.jcraft.Compression",
+                "com.jcraft.jsch.jcraft.Compression",
```
Comment (gastaldi) [inline extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java]: This line is repeated:

```suggestion
```

File: extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java [hunk 2]
```diff
@@ -29,7 +29,59 @@ ReflectiveClassBuildItem reflection() {
...
+                "org.eclipse.jgit.api.MergeCommand$FastForwardMode",
```
Comment (gastaldi) [inline extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java]: Can you group the `org.eclipse.jgit` classes and order the newly added items alphabetically?

**Review by gastaldi — CHANGES_REQUESTED** [review]
LGTM, added some minor comments. It would be nice to have a corresponding test case if possible

File: extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java (hunk 1 above)
Comment (andrejpetras) [reply extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java]: Done!

File: extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java (hunk 2 above)
Comment (andrejpetras) [reply extensions/jgit/deployment/src/main/java/io/quarkus/jgit/runtime/deployment/JGitProcessor.java]: Done!

**Comment by andrejpetras** [comment]
@gastaldi For the tests I need your help. Do we have any example how can I test it?

**Comment by gastaldi** [comment]
Hmm, for the tests I think it would probably require an embedded Git server with SSH enabled. I'm fine with this PR as is right now though

**Review by gastaldi — APPROVED** [review]

**Comment by gastaldi** [comment]
@andrejpetras can you squash your commits into a single one? That should be enough to merge this PR. Thanks!

**Comment by andrejpetras** [comment]
Squash commit: done!
For the test I use the TestContainers in the this project https://github.com/lorislab/fluh
https://github.com/lorislab/fluh/blob/master/src/test/java/org/lorislab/cd/fluh/AbstractTest.java

**Review by gsmet — APPROVED** [review]

**Pull request merged**